opentracing-spring-cloud/src/main/java/io/opentracing/contrib/spring/cloud/async/instrument/TracedExecutor.java

```diff
@@ -0,0 +1,28 @@
+package io.opentracing.contrib.spring.cloud.async.instrument;
+
+import java.util.concurrent.Executor;
+
+import io.opentracing.Tracer;
+import io.opentracing.contrib.concurrent.TracedRunnable;
+
+/**
+ * {@link Executor} that wraps {@link Runnable} in a
+ * {@link TracedRunnable} that propagates parent span inside the runnable.
+ *
+ * @author kameshsampath
+ */
+public class TracedExecutor implements Executor {
```

Comment:
This should really be in java-concurrent OT contrib.

Reply:
+1, I will create a PR